### PR TITLE
[CORL-462] Update the wordpress plugin to support Coral version 5

### PR DESCRIPTION
**Pull request by nick-funk**
This updates the plugin to allow switching between version 4 and 5.

Instructions for how to setup a local development environment to test this is available at:

https://app.tettra.co/teams/coralproject/pages/wordpress-development-environment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ if ( comments_open() ) {
 ```
 
 ## Version
-Talk version <= `v3.9.1` use talk-wp-plugin `v0.0.6`
+Coral version <= `v3.9.1` use plugin version `v0.0.6`
 
-Talk version >= `4.0.0` use talk-wp-plugin `v0.1.0`
+Coral version >= `4.0.0` use plugin version `v0.1.0`
+
+Coral version >= `5.0.0` use plugin version `v0.2.0`
```
Comment (nick-funk) [inline README.md:53]: @wyattjoh, I've indicated that there is a version bump for the plugin to support the v4/5 switching in this PR.

Please advise on what version number we want to use so I can update it in source as well as package this up and/or tag the version on master appropriately.

File: inc/comments-template.php [hunk 2]
```diff
@@ -11,12 +11,38 @@
...
+	<div class="<?php echo esc_attr( $talk_container_classes ); ?>" id="coral_thread"></div>
+	<script type="text/javascript">
+	(function() {
+		var d = document, s = d.createElement('script');
+		s.src = "<?php echo esc_url( $talk_url . '/assets/js/embed.js' ); ?>"
+		s.onload = function() {
+			Coral.createStreamEmbed({
+				id: "coral_thread",
+				autoRender: true,
+				rootURL: "<?php echo esc_url( $talk_url ); ?>",
```
Comment (nick-funk) [inline inc/comments-template.php:32]: @wyattjoh, we have a nice embed here that seems to work with my local dev environment.

Note that I haven't specified `storyID` or `storyURL`. Is this fine, or should I pull them from somewhere? Perhaps convenience functions are available in the PHP scripts that I'm not aware of?

File: inc/talk-settings-static-content.php [hunk 3]
```diff
@@ -8,20 +8,20 @@
 ?>
 <p>
 	<?php printf(
-		esc_html__( 'You can find out how to install and manage Talk %shere%s.', 'coral-project-talk' ),
+		esc_html__( 'You can find out how to install and manage Coral %shere%s.', 'coral-project-talk' ),
 		'<a href="https://docs.coralproject.net/talk/">',
```
Comment (nick-funk) [inline inc/talk-settings-static-content.php:12]: Changing up the names to `Coral` everywhere when the URL still points to `...something coral.../talk/` seems a bit weird.

Not sure if we want to fully go forward with the change of name to Coral with this plugin release, or wait until we've consolidated all document with our next release?

File: README.md (hunk 1 above)
Comment (wyattjoh) [reply README.md:53]: v0.2.0 is good as this isn't a breaking change to existing v4 installs as far as I know.

File: inc/comments-template.php [hunk 4]
```diff
@@ -11,12 +11,38 @@
...
+$talk_version = get_option( 'coral_talk_version' );
+
 $div_id = 'coral_talk_' . absint( rand() );
-if ( ! empty( $talk_url ) ) : ?>
+
+if ( empty( $talk_url) ):
+	exit();
+endif;
+
+if ( $talk_version == "4" ) : ?>
```
Comment (wyattjoh) [inline inc/comments-template.php]: For fallback, you should instead do something like `if ( $talk_version ~= "5" ) : ?>` instead, as those who have the plugin installed already should still load the correct template.

Comment (nick-funk) [reply inc/comments-template.php]: Excellent idea and good catch, will update shortly!

File: inc/talk-settings-static-content.php (hunk 3 above)
Comment (wyattjoh) [reply inc/talk-settings-static-content.php:12]: I'd just leave this for now until we get the docs handled, then it'll have to be managed across the board.

File: inc/comments-template.php (hunk 2 above)
Comment (wyattjoh) [reply inc/comments-template.php:32]: You could add [`get_the_ID()`](https://developer.wordpress.org/reference/functions/get_the_id/) to this as the `storyID` if you want I suppose!

File: inc/class-talk-settings-page.php [hunk 5]
```diff
@@ -145,6 +154,40 @@ public function render_container_classes_field() {
...
+					if (esc_attr( get_option( 'coral_talk_version' ) ) === "4")
+						echo "selected=\"selected\""
+				?>
+			>
+				4
+			</option>
+			<option value="5"
+				<?php 
+					if (esc_attr( get_option( 'coral_talk_version' ) ) === "5")
+						echo "selected=\"selected\""
```
Comment (wyattjoh) [inline inc/class-talk-settings-page.php:183]: I would suggest moving the `esc_attr( get_option( 'coral_talk_version' ) )` to the top of the `render_version_field` function if we can, then it should be good to merge!

File: inc/comments-template.php [hunk 6]
```diff
@@ -11,8 +11,34 @@
...
+				rootURL: "<?php echo esc_url( $talk_url ); ?>",
+				storyID: "<?php echo get_the_ID(); ?>",
+			});
+		};
+		(d.head || d.body).appendChild(s);
+	})();
+	</script>
+<?php endif;
+
+if ( $talk_version != "5" ) : ?>
```
Comment (wyattjoh) [inline inc/comments-template.php]: Why not `else`?

**Review by wyattjoh — CHANGES_REQUESTED** [review]

File: inc/comments-template.php (hunk 6 above)
Comment (nick-funk) [reply inc/comments-template.php]: Wanted to explain what "else" would be, in this case, everything prior to 5.
Will update and add comment.

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**